blocks/math/test/qa_ExpressionBlocks.cpp

```diff
@@ -0,0 +1,105 @@
+
+        expect(approx(source1.default_value, T(7), T(1e-3f)));
+        expect(approx(source2.default_value, T(5), T(1e-3f)));
+        expect(approx(exprBlock.param_a.value, T(3), T(1e-3f)));
+        expect(eq(tagSink._samples.size(), source1.n_samples_max));
+        expect(eq(tagSink._samples.size(), source2.n_samples_max));
+        expect(approx(tagSink._samples[0], T(42), T(1e-3f)));
+    } | std::tuple<float, double>{};
+
+    "ExpressionBulk"_test = []<typename T>(const T&) {
+        Graph graph;
+
+        auto& source    = graph.emplaceBlock<testing::CountingSource<T>>({{"n_samples_max", 10U}});
+        auto& exprBlock = graph.emplaceBlock<ExpressionBulk<T>>({{"expr_string", "vecOut := a * vecIn"}, {"param_a", T(2)}});
+        auto& tagSink   = graph.emplaceBlock<testing::TagSink<T, USE_PROCESS_ONE>>({{"log_samples", true}});
+
+        expect(eq(gr::ConnectionResult::SUCCESS, graph.connect<"out">(source).template to<"in">(exprBlock)));
+        expect(eq(gr::ConnectionResult::SUCCESS, graph.connect<"out">(exprBlock).template to<"in">(tagSink)));
+
+        auto sched = gr::scheduler::Simple<>(std::move(graph));
+        expect(sched.runAndWait().has_value());
+
+        expect(eq(tagSink._samples.size(), source.n_samples_max));
+        expect(approx(exprBlock.param_a.value, T(2), T(1e-3f)));
+        for (std::size_t i = 0; i < tagSink._samples.size(); ++i) {
+            T expected = T(1 + i) * T(2);
+            expect(approx(tagSink._samples[i], expected, T(1e-6))) << fmt::format("should be: output[{}] ({}) == {} * input[{}] ({}) ", i, tagSink._samples[i], exprBlock.param_a, i, expected);
+        }
+    } | std::tuple<float, double>{};
+
+    "ExpressionBulk - exceptions"_test = [](const bool enableERuntimeChecks) {
+        Graph graph;
+
+        auto&             source    = graph.emplaceBlock<testing::CountingSource<float>>({{"n_samples_max", 10U}});
+        const std::string exprStr   = R""(for (var i := 0; i < 100; i += 1) {
+   vecOut[i] := vecIn[i+1];
+}
+)"";
+        auto&             exprBlock = graph.emplaceBlock<ExpressionBulk<float>>({{"expr_string", exprStr}, {"runtime_checks", enableERuntimeChecks}});
+        auto&             tagSink   = graph.emplaceBlock<testing::TagSink<float, USE_PROCESS_ONE>>({{"log_samples", true}});
+
+        expect(eq(gr::ConnectionResult::SUCCESS, graph.connect<"out">(source).template to<"in">(exprBlock)));
+        expect(eq(gr::ConnectionResult::SUCCESS, graph.connect<"out">(exprBlock).template to<"in">(tagSink)));
+
+        auto sched = gr::scheduler::Simple<>(std::move(graph));
+
+        try {
+            expect(sched.runAndWait().has_value());
+            expect(false) << fmt::format("should have failed");
+        } catch (const gr::exception& ex) {
+            expect(true);
+            fmt::println("failed correctly with:\n{}\n", ex);
+        } catch (...) {
+            expect(false) << fmt::format("caught unknown/unexpected exception");
+        }
```

Comment:
Why not use ut's builtin exception matcher?
```suggestion
expect(throws<gr::exception>([] { expect(sched.runAndWait().has_value()); }))
```
https://github.com/boost-ext/ut/blob/master/README.md?plain=1#L1116